From: Elvis Angelaccio <elvis.angelaccio@kdemail.net>
Date: Wed, 23 Dec 2015 16:19:29 +0000
Subject: Fallback to read-only mode if there are no read-write executables
X-Git-Url: http://quickgit.kde.org/?p=ark.git&a=commitdiff&h=087e5aab49c60ac5930742fe892fa930048e2f43
---
Fallback to read-only mode if there are no read-write executables

Commit 2d000a0 introduced executables check when loading a plugin. However the
current behavior is too restrictive: if one wants only to open a rar or a
zip archive, there is no need to require also the rar or zip program to be
installed. Plus, some distributions (e.g. Archlinux) ship only unrar in their
official repositories.

With this commit, Ark is able to understand that e.g. unrar is installed but
rar is not. In this case, Ark can and should fallback to read-only mode,
to disable the Add/Delete actions in the toolbar.

BUG: 357057
FIXED-IN: 15.12.1

CC: rthomsen6@gmail.com
---


--- a/kerfuffle/archive_kerfuffle.cpp
+++ b/kerfuffle/archive_kerfuffle.cpp
@@ -173,6 +173,9 @@
 
             if (iface->findExecutables(!isReadOnly)) {
                 return new Archive(iface, isReadOnly, parent);
+            } else if (!isReadOnly && iface->findExecutables(false)) {
+                qCWarning(ARK) << "Failed to find read-write executables: falling back to read-only mode for read-write plugin" << pluginName;
+                return new Archive(iface, true, parent);
             } else {
                 qCWarning(ARK) << "Failed to find needed executables for plugin" << pluginName;
             }

